File: ObserverV2Java/ObserverJava/src/src/Subscriber.java
package src;

import include.IObserver;
import include.ISubscriber;

import java.util.ArrayList;

public class Subscriber implements ISubscriber {

    public Subscriber() {
        System.out.println("Subscriber::Subscriber");
    }

    public void attach(IObserver o) {
        observers.add(o);
    }

    public void detach(IObserver o) {
        observers.remove(o);
    }

    public ArrayList<IObserver> getSubscribers() {
        return observers;
    }

    private ArrayList<IObserver> observers = new ArrayList<IObserver>();
}

File: ObserverV2C#/Observer/src/Subscriber.cs
using Observer.include;
using System;
using System.Collections.Generic;
using System.Text;

namespace Observer.src
{
   public class Subscriber: ISubscriber {

        public Subscriber() {

        }

        public void Attach(IObserver o)
        {
            observers.Add(o);
        }
        public void Detach(IObserver o)
        {
            observers.Remove(o);
        }

        public List<IObserver> GetSubscribers()
        {
            return observers;
        }

        private List<IObserver> observers = new List<IObserver>();
    }
}
